### DISCUSSION TITLE
Question regarding env.sh

### DISCUSSION THREAD

**Discussion by kennyluk** [jjlin/vaultwarden-shared-hosting#2] [Q&A]
Hi, thanks for sharing this. I am using this to host a bitwarden vault on DH and works quite well.

Can I ask if the env.sh script is setting up the enviroment for the bitwarden_rs binary? I wanted to set the ROCKET_WORKERS and WEBSOCKET_ENABLED variables and wanted to know if this is the best way to do it.

Thanks.

kl

**Comment by kennyluk** [comment]
Hi, thanks for sharing this. I am using this to host a bitwarden vault on DH and works quite well.

Can I ask if the env.sh script is setting up the enviroment for the bitwarden_rs binary? I wanted to set the ROCKET_WORKERS and WEBSOCKET_ENABLED variables and wanted to know if this is the best way to do it.

Thanks.

kl

**Reply by jjlin** [reply]
Yes, the variables in `env.sh` will be set in the bitwarden_rs env. IIRC, bitwarden_rs defaults `ROCKET_WORKERS` to 10, and while I've never had a problem with that, you can change it if you really want.

As mentioned in the README, WebSockets are not supported in this implementation. You can set `WEBSOCKET_ENABLED`, but proxying WebSocket connections is not implemented in the WSGI app, so it won't work properly. I developed this setup for a friend, and I don't actually run on shared hosting myself, so I'm not really planning to do further development on this. I haven't looked into it at all, but I suspect it's not that hard to add WebSocket support. If you figure out how to do it, feel free to send a PR.

**Reply by kennyluk** [reply]
Thank you for taking the time to write back. I thought the use of WSGIProxy2 as a reverse proxy to wrap around bitwarden_rs was very clever to workaround the limitations at Dreamhost.

I am very new at this but will experiment with it and share what I learn.
